lib/selenium-cucumber/methods/click_elements_methods.rb

```diff
@@ -1,10 +1,23 @@
 require_relative 'required_files'
 
-
 def click(access_type, access_name)
   $driver.find_element(:"#{access_type}" => "#{access_name}").click
 end
 
+def click_by_text(access_type, access_name, text)
```

Comment:
no no, I am just adding verification that before entering loop, page should contain text then and only then check for particular element with that text.

for e.g.
I want to click on link with text "XYZ" from this page https://selenium-cucumber.github.io/
but there is no link with such text, with your logic it will first find all links and then one by one check its text.
But if you check at start that at-least text present somewhere in page. then you can continue finding elements.